### DISCUSSION TITLE
Blink-fs not installing in code

### DISCUSSION THREAD

**Discussion by chaloum** [blinksh/blink#1795] [Support]
I’m using the latest version of Blink and the latest iPadOS on a 5th generation iPad Pro (2021)

When launching Code form the Blink command line Bink says that Blink-fs needs to be installed from the marketplace (Extensions)

When I click on the install button, for a second or the buttons says that it’s installing but then the button text returns to install and  I get that error, Error while installing 'Blink-FS' extension.

any help in getting this installed would be appreciated.

Thanks
Marcel

**Comment by carloscabanero** [comment]
Hey Marcel. There should be no mystery to installing it. I just tried uninstalling and reinstalling in an iPad and did so in a clean iPhone as well. Could it be a hiccup from vscode.dev itself? Are you logged in as any specific user and maybe are there limitations there?

**Comment by chaloum** [comment]
Nope,  not logged in or signed in to anything including in Code.  I have no problems installing the extension when using the web version of VS-Code its only when launching Code in Blink Shell that I have issues installing the Blink-FS extension

**Reply by carloscabanero** [reply]
That's weird. If you do `code https://vscode.dev` can you install it there? From there you should be able to go the usual `code .` route.

Last resort would be to try reinstalling the app (make a backup of .blink first if you have important keys, etc...).

**Comment by chaloum** [comment]
Just to add on to this, if I try to install using the command  ext install blinkshellinc.blink-fs I get the error, XHR failed.  

I also not able to install any extension in the “market place”

**Comment by chaloum** [comment]
And to add more information to this.  If I switch over the to insiders edition of coder within Blink Shell,  and not the default stable version,  I can install extensions.

**Reply by carloscabanero** [reply]
Blink Code is just a layer on top of VS Code, but it does not change the core functionality. If it does not install it would be an issue on VS Code itself, not sure why one version works and not the other.

**Reply by chaloum** [reply]
As said /i installs fine when using the basic web version but doesn’t install what run in Blink shell

**Comment by carloscabanero** [comment]
Merging issues and a bunch of comment threads here, with everything we know about this. We have been unable to replicate from our side, so any help debugging is greatly appreciated. 

Here is what we know:
- Some Blink installations are having problems when installing extensions. It looks like it is any extension and not just Blink-FS.
- Extensions work when installed through regular Safari on vscode.dev
- A possible workaround is to use VS Code insiders instead: `code . --vscode-url "https://insiders.vscode.dev`

Here are some questions / things to try that would be helpful to get more information. Please report here after trying:
- Most immediate, is to connect to a Mac and Safari, to do a remote debugging of the Blink Code web window. The window should show some output when trying to install the extension with more information about what may be going on.
- Try loading VSCode through `browse vscode.dev` and install the app there. This shouldn't work, as it is the same browser, but there is a different browser configuration and maybe that is causing issues.
- As this seems to affect only a few number of Blink installations, resetting the app may work. Try to uninstall and reinstall the app. If you have keys, etc... please backup the .blink folder first. In the same manner, if you have access to TestFlight AND the AppStore versions it would be useful to see if it works on one and not the other.
- Are there any special settings like configuration sync turned on? 
- Stupid but not so stupid, sometimes on VS Code a tap is not the same as a click. Have you tried both?
- Reports online for similar issues usually blame it on the network. Given that this works on the normal browser, I would discard those, but if someone has time, please check:  https://stackoverflow.com/questions/37697039/unable-to-install-visual-studio-code-extensions

Some changes from our side that may be useful in the future:
- Restarting WKWebViews.

**Reply by deviantintegral** [reply]
I can answer for the one part of this. I’m a brand new user so I had no existing config. All I’ve done is add ssh keys and hosts. This is with a fully up to date iPadOS, is it possible this is a new bug in 16.5 or 16.5.1?

**Reply by carloscabanero** [reply]
Did you try by any chance to delete and reinstall Blink? That way we could discard that.

I don't think it is an issue with Blink or iOS version. It still works for a big majority and we have all 17 devices working too.

**Reply by deviantintegral** [reply]
I just tried with a fresh install on my iPhone (which had never had Blink installed) and it throws the same error.

**Reply by carloscabanero** [reply]
- If you try to install the extension running `code https://vscode.dev`, does it let you do it there?
- From reading StackOverflow threads of people with similar errors, two more things to try:
  - We tried "not to be logged in or synced", but actually the opposite may work. Have you tried to login to your GitHub account when before installing the extension?
  - Are you running any sort of Proxy, etc?

Blink Code is a thin layer on top of VS Code. It is really just a "fancy browser", for keyboard shortcuts and a connection on the back so it can communicate with Blink-FS. We are trying to think what else we may be able to do from our side.

Workaround for now is to use the insiders version, which hopefully will go live very soon and maybe fix this altogether.

**Reply by steverhall** [reply]
@carloscabanero

Extensions load when you start VS Code with `code . --vscode-url "https://vscode.dev"`. They also load if you start with `code . --vscode-url "https://insiders.vscode.dev"`, then use the settings cog to 'Switch to Stable Version'.

**Reply by deviantintegral** [reply]
`code https://vscode.dev` does not work.

`code . --vscode-url "https://vscode.dev"` does!

**Reply by steverhall** [reply]
Could this be related to the [pathOrUrl handling](https://github.com/blinksh/blink/blob/6723196f6b443b63051b4719b6e1090927eb8a10/Blink/Commands/code.swift#L124)? Extensions load when just using `code .` (without the --vscode-url argument), but not `code`.

**Reply by carloscabanero** [reply]
It may, although that doesn't explain why it works for most. When I read your last message I thought maybe they changed something in the way they enforce some routes and unload stuff. I will check the code and we may push a TestFlight for you all here to test this specifically, as I still cannot repro.

**Comment by carloscabanero** [comment]
Checking here. Any changes since they pushed July release?

**Reply by steverhall** [reply]
Still unable to install extension when just executing `code` on it's own, `code .` continues to work. Using v16.1.10.772

**Reply by revansdxr** [reply]
Confirmed... "code ." allows me to successfully install the blink-fs extension.   "code" by itself throws an error installing extensions.  Looks like this might be a pathing issue?
